src/lint/track_config.nim

```diff
@@ -579,19 +580,24 @@ proc satisfiesSecondPass(trackConfigContents: string; path: Path): bool =
   let trackConfig = toTrackConfig(trackConfigContents)
   result = true
 
-  let conceptSlugs = getConceptSlugs(trackConfig)
-  checkPractices(trackConfig.exercises.practice, conceptSlugs, result, path)
-  let conceptsTaught = checkExerciseConcepts(trackConfig, conceptSlugs, result,
-                                             path)
-  checkConceptExercisePrerequisites(trackConfig, conceptSlugs, conceptsTaught,
-                                    result, path)
-  checkPrerequisites(trackConfig.exercises.practice, conceptSlugs,
-                     conceptsTaught, result, path)
-  checkExercisesPCP(trackConfig.exercises.`concept`, result, path)
-  checkExercisesPCP(trackConfig.exercises.practice, result, path)
-  checkExerciseSlugsAndForegone(trackConfig.exercises, result, path)
-
-proc isValidTrackConfig(data: JsonNode; path: Path): bool =
+  let exercises = trackConfig.exercises
+  let conceptExercises = exercises.`concept`
+  let practiceExercises = exercises.practice
+  let concepts = trackConfig.concepts
+
+  let conceptSlugs = getConceptSlugs(concepts)
+  checkPractices(practiceExercises, conceptSlugs, result, path)
+  let conceptsTaught = checkExerciseConcepts(conceptExercises, conceptSlugs,
+                                             result, path)
+  checkPrerequisites(conceptExercises, conceptSlugs, conceptsTaught, result,
+                     path)
+  checkPrerequisites(practiceExercises, conceptSlugs, conceptsTaught, result,
+                     path)
+  checkExercisesPCP(conceptExercises, result, path)
+  checkExercisesPCP(practiceExercises, result, path)
+  checkExerciseSlugsAndForegone(exercises, result, path)
+
+proc satisfiesFirstPass(data: JsonNode; path: Path): bool =
```

Comment:
I'm not sure about this name. It explains how something is happening, but not why. I think we want that indicates that the type of the JSON object is checked.

Reply:
I don't like the name either. But the problem with something like `checkTypes` is that our first pass does a lot more than check the types - it's where most of our current linting logic is.

I didn't find a better name, so let's see if we can find a better name later. `satisfiesFirstPassThatIncludesATypeCheck` is a bit verbose :)

In the meantime, I've added a comment to help a little.